source/lac/petsc_matrix_base.cc

```diff
@@ -34,11 +34,14 @@ namespace PETScWrappers
     MatrixBase::const_iterator::Accessor::
     visit_present_row ()
     {
-      // if we are asked to visit the
-      // past-the-end line, then simply
-      // release all our caches and go on
-      // with life
-      if (this->a_row == matrix->m())
+      // if we are asked to visit the past-the-end line (or a line that is not
+      // stored on the current processor), then simply release all our caches
+      // and go on with life
+      if ((this->a_row == matrix->m())
```

Comment:
This condition should now be redundant, right? No matrix is going to store the `m`th row.

Reply:
I suspect that to be true -- I had left it in because I wasn't quite sure whether `local_range.second` is always at most equal to `m()`. But it must be, and if that's not the case, then there's clearly a bug here. So I'll rewrite this condition.